ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,113 @@
+__all__ = ["MeanAveragePrecision"]
+
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+
+try:
+    from torchvision.ops import box_iou
+except ImportError:
+    raise RuntimeError("This module requires torchvision to be installed.")
```

Comment:
@louis-she please put this into `MeanAveragePrecision` constructor. Otherwise, while imported in `__init__` it will directly fail if no torchvision.